### PR TITLE
test: add e2e test cases for route with log plugin

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [x] E2E test

- Related issues

#636

### CODE REVIEW

**Comment by membphis** [comment]
ci failed

File: api/test/e2e/route_with_log_plugin_test.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package e2e
+
+import (
+	"github.com/stretchr/testify/assert"
```
Comment (idbeta) [inline api/test/e2e/route_with_log_plugin_test.go]: The third-party library should placed below.

**Review by idbeta — COMMENTED** [review]
The third-party library is placed below

File: api/test/e2e/route_with_log_plugin_test.go (hunk 1 above)
Comment (nic-chen) [reply api/test/e2e/route_with_log_plugin_test.go]: fixed.

**Review by gxthrj — APPROVED** [review]

File: api/test/docker/apisix_config.yaml [hunk 2]
```diff
@@ -38,3 +38,6 @@ apisix:
       enable: true
       ssl_cert: "/usr/local/apisix/certs/apisix.crt"
       ssl_cert_key: "/usr/local/apisix/certs/apisix.key"
+
+nginx_config:
```
Comment (membphis) [inline api/test/docker/apisix_config.yaml:42]: I don't think we need to change this

File: api/test/e2e/route_with_log_plugin_test.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+			ExpectStatus: http.StatusNotFound,
+			ExpectBody:   `{"error_msg":"404 Route Not Found"}`,
+		},
+		{
+			caseDesc: "create route",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/opentracing",
```
Comment (membphis) [inline api/test/e2e/route_with_log_plugin_test.go:44]: why not use /hello ?

File: api/test/e2e/route_with_log_plugin_test.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+	for _, tc := range tests {
+		testCaseCheck(tc)
+	}
+
+	// sleep for process log
+	time.Sleep(1500 * time.Millisecond)
+
+	// verify http logger by checking log
+	bytes, err := ioutil.ReadFile("../docker/apisix_logs/error.log")
```
Comment (membphis) [inline api/test/e2e/route_with_log_plugin_test.go:89]: That is a not good idea. We should use a fake upstream for confirming if we got the log data.

File: api/test/e2e/route_with_log_plugin_test.go [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+	for _, tc := range tests {
+		testCaseCheck(tc)
+	}
+
+	// sleep for process log
+	time.Sleep(1500 * time.Millisecond)
+
+	// verify http logger by checking log
+	bytes, err = ioutil.ReadFile("../docker/apisix_logs/error.log")
```
Comment (membphis) [inline api/test/e2e/route_with_log_plugin_test.go:146]: Ditto.

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/e2e/route_with_log_plugin_test.go (hunk 4 above)
Comment (membphis) [reply api/test/e2e/route_with_log_plugin_test.go:90]: If we check it in the log, I think we should follow those steps:

Clean the log, run the test, and then make a check.

We need to confirm the content of log is the one we expected.

**Review by membphis — COMMENTED** [review]
Add more steps

**Comment by nic-chen** [comment]
@membphis updated.

**Review by membphis — APPROVED** [review]

**Pull request merged**